src/snowflake/snowpark/table.py

```diff
@@ -314,26 +336,41 @@ def __init__(
         self.is_cached: bool = self.is_cached  #: Whether the table is cached.
         self.table_name: str = table_name  #: The table name
         self._is_temp_table_for_cleanup = is_temp_table_for_cleanup
+        self._time_travel_config = time_travel_config
 
         # By default, the set the initial API call to say 'Table.__init__' since
         # people could instantiate a table directly. This value is overwritten when
         # created from Session object
         set_api_call_source(self, "Table.__init__")
 
     def _copy_without_ast(self):
+        kwargs = {}
+        if self._time_travel_config:
+            time_travel_dict = self._time_travel_config._asdict()
+            time_travel_dict["time_travel_mode"] = time_travel_dict.pop("mode")
+            kwargs.update(time_travel_dict)
+
         return Table(
             self.table_name,
             session=self._session,
             is_temp_table_for_cleanup=self._is_temp_table_for_cleanup,
             _emit_ast=False,
+            **kwargs,
         )
 
     def __copy__(self) -> "Table":
+        kwargs = {}
+        if self._time_travel_config:
+            time_travel_dict = self._time_travel_config._asdict()
+            time_travel_dict["time_travel_mode"] = time_travel_dict.pop("mode")
```

Comment:
can you override the _asdict method in `TimeTravelConfig` since it is repeated here